src/test/java/org/pesmypetcare/webservice/dao/medalmanager/UserMedalDaoTest.java

```diff
@@ -0,0 +1,111 @@
+package org.pesmypetcare.webservice.dao.medalmanager;
+
+import com.google.cloud.firestore.WriteBatch;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.pesmypetcare.webservice.entity.medalmanager.UserMedalEntity;
+import org.pesmypetcare.webservice.error.DatabaseAccessException;
+import org.pesmypetcare.webservice.error.DocumentException;
+import org.pesmypetcare.webservice.thirdpartyservices.adapters.firestore.FirestoreCollection;
+import org.pesmypetcare.webservice.thirdpartyservices.adapters.firestore.FirestoreDocument;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.same;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.willDoNothing;
+import static org.mockito.Mockito.verify;
+
+/**
+ * @author Oriol Catalán
+ */
+@ExtendWith(MockitoExtension.class)
+public class UserMedalDaoTest {
+    private static final String OWNER = "OwnerUsername";
+    private static final String OWNER_ID = "OwnerId";
+    private static final String USER_MEDAL_NAME = "UserMedalName";
+    private static final String SIMPLE_FIELD = "progress";
+    private static final Double VALUE = 2.0;
+
+    private static UserMedalEntity userMedalEntity;
+    private static List<Map<String, Object>> userMedalList;
+
+
+    @Mock
+    private FirestoreCollection dbCol;
+    @Mock
+    private FirestoreDocument dbDoc;
+    @Mock
+    private WriteBatch batch;
+
+    @InjectMocks
+    private UserMedalDao userMedalDao = new UserMedalDaoImpl();
+
+    @BeforeEach
+    public void setUp() {
+        userMedalEntity = new UserMedalEntity("Walker", 6.0, 2.0, new ArrayList<String>());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['6.0' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5517612)